Tests/XCDiffCoreTests/Library/URLExtensionsTests.swift

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright 2019 Bloomberg Finance L.P.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+@testable import XCDiffCore
+import XCTest
+
+final class URLExtensionsTests: XCTestCase {
+    func test_relative_path_longer() {
+        XCTAssertEqual(generateRelative("/usr/bin/agent/demo.gz", "/usr/bin"), "agent/demo.gz")
+        XCTAssertEqual(generateRelative("/user/db/demo.exe", "/user/tb"), "../db/demo.exe")
+    }
+
+    func test_empty_path() {
+        XCTAssertEqual(generateRelative("/", "/"), "")
+    }
+
+    func test_relative_path_shorter() {
+        XCTAssertEqual(generateRelative("/user/x/y", "/user/x/y/a/b/c"), "../../..")
+    }
+
+    func generateRelative(_ first: String, _ second: String) -> String? {
```

Comment:
`private`? Also, maybe renaming it to `relative(...` or `computeRelative` would work better here?